### PR TITLE
Handle Redis connection failure gracefully.

### LINKED ISSUE #97: Extend $fail_gracefully to other methods rather than only constructor
**Context:**

The `WP_Object_Cache` class takes `$fail_gracefully` variable as parameter in its constructor. And the Redis Object Cache fails gracefully without throwing any exception if it fails to connect to Redis.

But this does not extend to other methods i.e., `add`, `replace`, `add_or_replace`, `delete`, `get`, `set` etc.

This is fine when the Redis connectivity is checked in the constructor. But let's say if Redis connectivity is lost for some unknown reasons (network failures, Redis server failures etc.) at later stage after constructor is already executed, the plugin will throw an exception and the php process will terminate abruptly.

**Proposal:**
Store the `$fail_gracefully` variable as private class variable. And respect its value in other methods as well along with the constructor such as `add`, `replace`, `set`, `get` etc.

With this, all the methods of `WP_Object_Cache` class will fail gracefully in case of an error.

Open for other suggestions. This scenario was exposed in the production environment of one of the project that I'm working on.

### PR DESCRIPTION
**Pull request by desaiuditd**
Fixes #97

### CODE REVIEW

File: includes/object-cache.php [hunk 1]
```diff
@@ -1265,6 +1321,22 @@ protected function is_serialized($data, $strict = true)
...
         return false;
     }
+
+    /**
+     * Handle the redis failure gracefully or throw an exception.
+     *
+     * @param \Exception $exception Exception thrown.
+     */
+    private function handle_redis_failure($exception) {
+        // When Redis is unavailable, fall back to the internal back by forcing all groups to be "no redis" groups
```
Comment (aprea) [inline includes/object-cache.php]: I know this was an existing comment that was simply moved but is `internal back` a typo?

Comment (desaiuditd) [reply includes/object-cache.php]: Fixed - 141b0f7

**Comment by tillkruss** [comment]
I think we should log the silent exception.

**Comment by tillkruss** [comment]
@desaiuditd: What do you reckon about returning early? https://github.com/tillkruss/redis-cache/commit/66431a5c4dcfcd5b28c0fb36860628e5f9a0cd43

**Comment by desaiuditd** [comment]
> @desaiuditd: What do you reckon about returning early? [66431a5](https://github.com/tillkruss/redis-cache/commit/66431a5c4dcfcd5b28c0fb36860628e5f9a0cd43)

Yeah. Looks legit. Explicit early return.

**Pull request merged**